### ISSUE TITLE
SMB3 client on samba server

### ISSUE THREAD

**Issue by hagaygo** [TalAloni/SMBLibrary#253]
Hi

My app works just fine using this great library when connecting samba servers for a long time.

As mentioned on https://github.com/TalAloni/SMBLibrary/issues/55 I understand the library should allow a client to connect using smb3 using SMB2Client class.

Changed the samba server to accept only SMB3 connection (using server min protocol = SMB3) and my app cant connect.

The Connection method returns false.

Setting the samba server back to allow smb2 and app is working again.

Does the library support such scenario (on the referenced issue it stated that it tries to perform a negotiation on SMB protocol version) ?

**Comment by hagaygo** [comment]
My app runs on android devices , so wireshark on the client might be tricky.

Is it enough to run it on server only ?

**Comment by TalAloni** [comment]
Sure it is, you'll capture all the requests made by the client

**Comment by hagaygo** [comment]
I found that also a linux client (under debian) shows similar problem.

When i set the samba server to
````
server min protocol = SMB2_10
````

Both smblibrary and linux cifs moun works find (linux mount also works with vers=3.0 parameter)

When i set the samba server to

````
server min protocol = SMB3
````

or

````
server min protocol = SMB3_00
````

Both smblibrary (as previous described) and linux mount does not work

dmesg shows the following error for the linux client :

````
 Dialect not supported by server. Consider  specifying vers=1.0 or vers=2.0 on mount for accessing older servers
````

On every case the share works fine on windows client (at least on windows 11 and 10)

So, To sum up , i think its a samba server issue or something i am missing in my configuration.

I'll try to ask for support over samba project to clear this issue up.

Feel free to close the issue if you got no interest in investigating this yourself.

**Comment by SeanSong2016** [comment]
Hi @hagaygo , did you solve the problem?
